### PR TITLE
editoast: refactor study tests

### PR DESCRIPTION
**Pull request by Wadjetz**
part of https://github.com/OpenRailAssociation/osrd/issues/6980

### CODE REVIEW

**Review by hamz2a — APPROVED** [review]

File: editoast/src/modelsv2/study.rs [hunk 1]
```diff
@@ -85,74 +85,71 @@ fn dates_in_order(a: Option<Option<NaiveDate>>, b: Option<Option<NaiveDate>>) ->
...
 
     use super::*;
-    use crate::fixtures::tests::db_pool;
-    use crate::fixtures::tests::study_fixture_set;
-    use crate::fixtures::tests::StudyFixtureSet;
-    use crate::fixtures::tests::TestFixture;
-    use crate::modelsv2::DbConnectionPool;
+    use crate::modelsv2::fixtures::create_project;
+    use crate::modelsv2::fixtures::create_study;
+    use crate::modelsv2::DbConnectionPoolV2;
```
Comment (leovalais) [inline editoast/src/modelsv2/study.rs:93]: `pretty_assertions`

File: editoast/src/modelsv2/study.rs [hunk 2]
```diff
@@ -85,74 +85,71 @@ fn dates_in_order(a: Option<Option<NaiveDate>>, b: Option<Option<NaiveDate>>) ->
...
-
-        // Create second study
-        let study_2 = study
-            .model
-            .clone()
-            .into_changeset()
-            .name(study.model.name.clone() + "_bis");
-
-        let _: TestFixture<Study> = TestFixture::create(study_2, db_pool.clone()).await;
+    async fn sort_project() {
```
Comment (leovalais) [inline editoast/src/modelsv2/study.rs]:
```suggestion
    async fn sort_study() {
```

File: editoast/src/modelsv2/study.rs [hunk 3]
```diff
@@ -85,74 +85,71 @@ fn dates_in_order(a: Option<Option<NaiveDate>>, b: Option<Option<NaiveDate>>) ->
...
-            project: _project,
-        } = study_fixture_set.await;
-
-        // Delete the study
-        let conn = &mut db_pool.get().await.unwrap();
-        Study::delete_static(conn, study.id()).await.unwrap();
-
-        // Second delete should fail
-        assert!(!Study::delete_static(conn, study.id()).await.unwrap());
+    async fn study_creation() {
```
Comment (leovalais) [inline editoast/src/modelsv2/study.rs]: This test is already included into `study_retrieve`. I'd say it's fine to remove it.

File: editoast/src/views/study.rs [hunk 4]
```diff
@@ -472,85 +441,172 @@ pub mod test {
...
+                "name": study_name,
+                "budget": study_budget,
+            }))
+            .to_request();
+        let response: StudyResponse = call_and_read_body_json(&app.service, request).await;
+
+        assert_eq!(response.study.name, study_name);
+        assert_eq!(response.study.budget, Some(study_budget));
+        assert_eq!(response.study.project_id, created_project.id);
+        assert_eq!(response.project.id, created_project.id);
```
Comment (leovalais) [inline editoast/src/views/study.rs]: Probably better `retrieve` and check that both the project and the study are equal to what's been returned by the response. 
(We could build the response within a transaction where all db operations have been performed, forget to commit it and return the response anyway. All these checks would hold but the study wouldn't have been patched.)

File: editoast/src/views/study.rs [hunk 5]
```diff
@@ -472,85 +441,172 @@ pub mod test {
...
+        assert_ne!(
+            response.project.last_modification,
+            created_project.last_modification
+        )
```
Comment (leovalais) [inline editoast/src/views/study.rs]:
```suggestion
        assert!(project.last_modification > created_project.last_modification);
        assert!(study > created_study.last_modification);
```

File: editoast/src/views/study.rs [hunk 6]
```diff
@@ -411,58 +409,29 @@ async fn list(
 
 #[cfg(test)]
 pub mod test {
-    use actix_http::Request;
     use actix_web::http::StatusCode;
+    use actix_web::test::call_and_read_body_json;
```
Comment (leovalais) [inline editoast/src/views/study.rs:414]: `pretty_assertions`

**Review by leovalais — CHANGES_REQUESTED** [review]
Thanks for this PR :) A few changes:

File: editoast/src/views/study.rs (hunk 5 above)
Comment (Wadjetz) [reply editoast/src/views/study.rs]: The API doesn't update last_modification on the study. Is it a bug, and should it update it?

Comment (leovalais) [reply editoast/src/views/study.rs]: I didn't?? :o

Then yes it's definitely a bug 🐛

**Review by leovalais — APPROVED** [review]
Tested. Thanks for the improvements 🚀

**Pull request merged**